docs/data/date-pickers/calendar-systems/AdapterHijri.tsx

```diff
@@ -2,12 +2,22 @@ import * as React from 'react';
 import moment from 'moment-hijri';
 import { AdapterMomentHijri } from '@mui/x-date-pickers/AdapterMomentHijri';
 import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
-import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+import { DateTimePicker } from '@mui/x-date-pickers/DateTimePicker';
+import { createTheme, ThemeProvider } from '@mui/material/styles';
+
+const theme = createTheme({ direction: 'rtl' });
```

Comment:
This causes issues when using a dark docs page theme. 🙈 
![Screenshot 2023-04-05 at 12 51 31](https://user-images.githubusercontent.com/4941090/230046364-003f021a-c4ad-4425-80ee-43db37be134c.png)

I propose to use the existing theme and extend it. Or feel free to come up with a better solution. 👌 
```diff
diff --git a/docs/data/date-pickers/calendar-systems/AdapterJalali.tsx b/docs/data/date-pickers/calendar-systems/AdapterJalali.tsx
index b869ec8ea..2a9ede08a 100644
--- a/docs/data/date-pickers/calendar-systems/AdapterJalali.tsx
+++ b/docs/data/date-pickers/calendar-systems/AdapterJalali.tsx
@@ -1,12 +1,16 @@
 import * as React from 'react';
+import useTheme from '@mui/system/useTheme';
 import { AdapterDateFnsJalali } from '@mui/x-date-pickers/AdapterDateFnsJalali';
 import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
 import { DateTimePicker } from '@mui/x-date-pickers/DateTimePicker';
 import { createTheme, ThemeProvider } from '@mui/material/styles';
 
-const theme = createTheme({ direction: 'rtl' });
-
 export default function AdapterJalali() {
+  const existingTheme = useTheme();
+  const theme = React.useMemo(
+    () => createTheme({ direction: 'rtl' }, existingTheme),
+    [existingTheme],
+  );
   return (
     <ThemeProvider theme={theme}>
       <div dir="rtl">
```

Reply:
Nice catch. I went to the [docs solution](https://mui.com/material-ui/customization/theming/#nesting-the-theme) providing a function to the `theme` props